src/utils/prettifier.ts

```
@@ -0,0 +1,370 @@
+
+    const nextNonWhiteSpaceCharPos = findNextNonWhitespaceChar(uglyString, index);
+    state.result += uglyString.substring(state.copyBeginOffset, index + 1);
+    state.copyBeginOffset = nextNonWhiteSpaceCharPos;
+
+    const isDanglingComma = ["}",
+        ")",
+        "]"].includes(uglyString[nextNonWhiteSpaceCharPos] ?? "");
+    const shouldBreak = !isDanglingComma && (
+        0 === state.bracketLevelCounter ||
+        ["{",
+            "("].includes((state.regionStack[state.regionStack.length - 1] ?? {}).type ?? ""));
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)